src/omega_edit/omega_edit.ts

```diff
@@ -127,11 +84,69 @@ export class OmegaEdit {
     }
   }
 
+  async search(
+    fileSize: number,
+    searchPattern: string
+  ): Promise<Array<number>> {
+    return await omegaEditSession.searchSession(
+      this.sessionId,
+      Buffer.from(searchPattern),
+      true,
+      0,
+      0,
+      fileSize
+    )
+  }
+
+  async replace(
+    sessionId: string,
+    offset: number,
+    len: number,
+    data: string = ''
+  ) {
+    await omegaEditChange.del(sessionId, offset, len)
+
+    // if no data sent don't insert
+    if (data !== '') {
+      await omegaEditChange.insert(sessionId, offset, data)
+    }
+  }
+
+  // Perform search getting on a single result
+  async singleSearch(fileSize: number, searchPattern: string): Promise<number> {
+    var result = await this.search(fileSize, searchPattern)
```

Comment:
To optimize this, set the limit to 1 so the server can stop searching after it finds the first instance of the pattern.